Disable test_nccl for ROCm

pritamdamania87 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#38801 Disable test_nccl for ROCm**

NCCL specific tests that shouldn't be run on ROCm

Differential Revision: [D21667348](https://our.internmc.facebook.com/intern/diff/D21667348/)

pritamdamania committed:
Disable test_nccl for ROCm

NCCL specific tests that shouldn't be run on ROCm

Differential Revision: [D21667348](https://our.internmc.facebook.com/intern/diff/D21667348/)

[ghstack-poisoned]

mruberry commented on test/distributed/test_nccl.py:
```diff
@@ -17,6 +18,7 @@
     TestCase = object  # noqa: F811
 
 
+@skipCUDAIfRocm
```
I don't think this will work. You should import TEST_WITH_ROCM from torch.testing._internal.common_utils and annotate each failing test:

```
@unittest.skipIf(TEST_WITH_ROCM, 'Test is failing on ROCm, see issue <link to issue here>')
```

mruberry reviewed (changes requested):
See comment. Also, we should link the associated issue for the flaky tests here (and in the code). 

cc @ngimel

pritamdamania committed:
Update on "Disable test_nccl for ROCm"


NCCL specific tests that shouldn't be run on ROCm

Differential Revision: [D21667348](https://our.internmc.facebook.com/intern/diff/D21667348/)

[ghstack-poisoned]

mruberry commented on test/distributed/test_nccl.py:
```diff
@@ -19,12 +21,14 @@
 
 class TestNCCL(TestCase):
 
+    @unittest.skipIf(TEST_WITH_ROCM, 'Skip NCCL tests for ROCm')
```
nit: link the issue in the string

pritamdamania87 replied:
Which issue should I link to here?

mruberry replied:
You should file an issue that these tests are failing on ROCm, label it with "topic: flaky-tests", and then link that issue.

mruberry replied:
(Make sure to copy the logs demonstrating these tests are failing on ROCm into the issue you create.)

mruberry replied:
(see https://fb.quip.com/24NyAwEsCd9a#QEJACADUdxe)

mruberry reviewed (approved):
I added a nit about adding a link to the issue for this in the code.

Otherwise I think this looks good. We should let the ROCm CI run to completion to verify.

pritamdamania committed:
Update on "Disable test_nccl for ROCm"


NCCL specific tests that shouldn't be run on ROCm

Differential Revision: [D21667348](https://our.internmc.facebook.com/intern/diff/D21667348/)

[ghstack-poisoned]
